### PR TITLE
Add kind/regression label

### PR DESCRIPTION
**Pull request by liggitt**
This allows labeling issues/PRs that are reporting/fixing regressions from previous releases, to aid in metrics around release quality.

cc @kubernetes/sig-contributor-experience @kubernetes/sig-release @justaugustus 

--
(edit by @spiffxp)
Lazy consensus timeout set for Monday, April 6, EOD US ET.

### CODE REVIEW

File: label_sync/labels.md [hunk 1]
```diff
@@ -98,7 +99,7 @@ larger set of contributors to apply/remove them.
 | <a id="sig/multicluster" href="#sig/multicluster">`sig/multicluster`</a> | Categorizes an issue or PR as relevant to sig-multicluster. <br><br> This was previously `sig/federation`, `sig/federation (deprecated - do not use)`, | anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
 | <a id="sig/network" href="#sig/network">`sig/network`</a> | Categorizes an issue or PR as relevant to sig-network.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
 | <a id="sig/node" href="#sig/node">`sig/node`</a> | Categorizes an issue or PR as relevant to sig-node.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
-| <a id="sig/onprem" href="#sig/onprem">`sig/onprem`</a> | REMOVING. This will be deleted after 2019-12-17 00:00:00 +0000 UTC <br><br> Categorizes an issue or PR as relevant to sig-onprem.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
+| <a id="sig/onprem" href="#sig/onprem">`sig/onprem`</a> | REMOVING. This will be deleted after 2019-12-17 00:00:00 +0000 +0000 <br><br> Categorizes an issue or PR as relevant to sig-onprem.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
```
Comment (justaugustus) [inline label_sync/labels.md:102]: There's a bug in the generator script causing this date to get mangled.
We can probably just drop this label given the note:
> REMOVING. This will be deleted after 2019-12-17 00:00:00 +0000 +0000 <br><br> Categorizes an issue or PR as relevant to sig-onprem.

Comment (liggitt) [reply label_sync/labels.md:102]: there are other instances of mangled deleteAfter dates... I'll leave that cleanup in the generator for a follow up

Comment (liggitt) [reply label_sync/labels.md:102]: >  I'll leave that cleanup in the generator for a follow up

or... maybe I won't... hooray for inconsistent generated results:

```
Validating labels...
Updating labels...
< unexpected
> missing
102c102
< | <a id="sig/onprem" href="#sig/onprem">`sig/onprem`</a> | REMOVING. This will be deleted after 2019-12-17 00:00:00 +0000 +0000 <br><br> Categorizes an issue or PR as relevant to sig-onprem.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
---
> | <a id="sig/onprem" href="#sig/onprem">`sig/onprem`</a> | REMOVING. This will be deleted after 2019-12-17 00:00:00 +0000 UTC <br><br> Categorizes an issue or PR as relevant to sig-onprem.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |

ERROR: labels.md out of date. Fix with hack/update-labels.sh
```

Comment (liggitt) [reply label_sync/labels.md]: removed

**Comment by saschagrunert** [comment]
Are regressions a subset of bugs?

**Comment by justaugustus** [comment]
/lgtm
/approve
/hold

This needs to be proposed to the community and held for lazy consensus timeout since it's a global change. I think 72 hours is probably fine.

/assign @cblecker @spiffxp

**Comment by liggitt** [comment]
> Are regressions a subset of bugs?

Yes, like kind/flake is a subset of kind/failing-test

**Comment by spiffxp** [comment]
/approve
but agree lazy consensus since this impacts all repos

**Comment by MorrisLaw** [comment]
This is a good idea!
/lgtm

**Comment by justaugustus** [comment]
Sent a note to k-dev, SIG ContribEx, and SIG Release: https://groups.google.com/d/topic/kubernetes-dev/dNuFIxNCjb4/discussion

Lazy consensus timeout set for Monday, April 6, EOD US ET.

**Comment by nikhita** [comment]
Lgtm :+1:

**Comment by hasheddan** [comment]
:+1:

**Comment by thockin** [comment]
I find the way "kind" labels work to be awkward.

In https://github.com/kubernetes/enhancements/pull/1554#issuecomment-592058697 I argue for a different schema, but maybe it's just too far gone to be fixed?

"""
IMO "less is more" means "less semantic complexity" vs "literally less
labels". Clearly orthogonal sets of labels are easy to use right and hard to
use wrong. Even now, kind/* sort of fails the orthogonality test. Think about
kind/api-change - can something be a bug and an API change? Or is
kind/api-change just for PRs while kind/bug is for issues and PRs?

To follow my "less is more" meaning, kind/* might be broken down into:

kind/bug
issues: something that is supposed to work but doesn't
PRs: a link to a kind/bug issue is strongly encouraged
kind/feature
issues: something that should be supported but isn't
PRs: a link to an enhancement is strongly encouraged
kind/test
issues: a test is flaking, failing, incomplete, or missing
PRs: test-related changes
kind/cleanup
issues: not really a bug or feature, but still needs work done
PRs: non-functional changes (refactoring, cleanup)
In general you would only need ONE of these on each issue and PR. The meanings
are clear and while some PRs serve 2 purposes, there's kind of a hierarchy. A
PR which adds a feature SHOULD come with tests, but is fundamentally a feature
change. A PR to fix tests MIGHT come with cleanups, but is fundamentally a
test fix. A bug MUST NOT come with a feature.

Heck, I could maybe be convinced that only kind/bug and kind/feature matter.

All of the other kind/* labels should move to something like flag/* - they
are flags on a PR or issue. "This is an API change", "this is a test flake",
"this is documentation". They are orthogonal to the fundamental categorization of
kind/*.

**Comment by dims** [comment]
I'd support this as a tactical option. Can sig-contribex folks please look at expanding the discussion that @thockin started above?

**Comment by liggitt** [comment]
I wouldn't object to regrouping all but the "primary" kind labels to some other prefix in a follow-up.

I see "previously" labels in the config file... does that allow renaming a label in the future, or does it just render help text?

**Comment by nikhita** [comment]
Regrouping labels as a follow-up :+1: 

> I see "previously" labels in the config file... does that allow renaming a label in the future, or does it just render help text?

It allows renaming labels in the future, and also renders the help text.

File: label_sync/labels.md [hunk 2]
```diff
@@ -98,7 +99,6 @@ larger set of contributors to apply/remove them.
 | <a id="sig/multicluster" href="#sig/multicluster">`sig/multicluster`</a> | Categorizes an issue or PR as relevant to sig-multicluster. <br><br> This was previously `sig/federation`, `sig/federation (deprecated - do not use)`, | anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
 | <a id="sig/network" href="#sig/network">`sig/network`</a> | Categorizes an issue or PR as relevant to sig-network.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
 | <a id="sig/node" href="#sig/node">`sig/node`</a> | Categorizes an issue or PR as relevant to sig-node.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
-| <a id="sig/onprem" href="#sig/onprem">`sig/onprem`</a> | REMOVING. This will be deleted after 2019-12-17 00:00:00 +0000 UTC <br><br> Categorizes an issue or PR as relevant to sig-onprem.| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
```
Comment (sftim) [inline label_sync/labels.md:101]: @liggitt I'd separate this removal into its own commit (or, if this repo uses _merge-strategy-squash_, its own PR). That makes a revert easier if needed.

Comment (liggitt) [reply label_sync/labels.md:101]: done

**Comment by saschagrunert** [comment]
I initially saw kinds as indicator for developers with respect to group PRs and issues to sort or filter them.

Now I see them more as hint for users since we transfer them into the release notes.
(so +1 for kind/regression)

This means that I think that kind/deprecation for example makes sense too, probably also a kind/dependency-change (not yet existing in k/k). But for example kind/cleanup seems to be the last resort if no other kind applies. 🙂

**Comment by spiffxp** [comment]
As far as reorganizing kind labels goes, https://github.com/kubernetes/community/issues/2032 is the last time we did this, migrating the org from ~14 kind labels to  7 +/- 2.  The idea was `area/foo` labels could add more detail from there.  I would be in favor of fewer labels.

**Comment by nikopen** [comment]
/lgtm
/approve

+100 on @thockin 's idea as IMO it is a clear improvement with no drawbacks.

An example is failing-tests and flakes, right now there's inconsistency with some flakes only having kind/flake, some both.  Having all as 'kind/failing-test' followed by 'flag/flake' where necessary would be more consistent. Not a big issue by any means, but the proposal is an improvement.

**Comment by justaugustus** [comment]
Lazy consensus timeout achieved. Removing hold.
/hold cancel
